agent/transformers/nagios.go

```diff
@@ -0,0 +1,86 @@
+package transformers
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/sensu/sensu-go/types"
+)
+
+// NagiosList contains a list of Nagios metrics
+type NagiosList []Nagios
+
+// Nagios contains values of Nagios performance data metric
+type Nagios struct {
+	Label     string
+	Value     float64
+	Timestamp int64
+}
+
+// Transform transforms a metric in Nagio perfdata format to Sensu Metric Format
+func (n NagiosList) Transform() []*types.MetricPoint {
+	var points []*types.MetricPoint
+	for _, nagios := range n {
+		mp := &types.MetricPoint{
+			Name:      nagios.Label,
+			Value:     nagios.Value,
+			Timestamp: nagios.Timestamp,
+			Tags:      []*types.MetricTag{},
+		}
+		points = append(points, mp)
+	}
+	return points
+}
+
+// ParseNagios parses a Nagios perfdata string into a slice of Nagios struct
+func ParseNagios(event *types.Event) (NagiosList, error) {
+	nagiosList := NagiosList{}
+
+	// Ensure we have some perfdata metrics and not only human-readable text
+	output := strings.Split(event.Check.Output, "|")
+	if len(output) != 2 {
+		return nil, errors.New("nagios perfdata format requires at least one performance data metric")
+	}
+
+	// Fetch the perfdata and remove leading & trailing whitespaces
+	perfdata := strings.Trim(output[1], " ")
+
+	// Split the perfdata into a slice of metrics
+	metrics := strings.Split(perfdata, " ")
+
+	// Create a Nagios metric for each perfdata metrics
+	for _, metric := range metrics {
+		// Clear everything after ';' and split the label and the value
+		parts := strings.Split(metric, ";")
```

Comment:
I can't find anything in the nagios documentation about this so sorry if I'm missing something, but what does `;` specify and why are we ignoring what comes after it?

Reply:
To be honest, I have absolutely no idea, I just blindly copied this logic from https://github.com/sensu/sensu-enterprise/blob/057f5af0d4960b70241fdf5181ce604d3afb4262/lib/sensu/enterprise/extensions/metric_parsers.rb#L55 😆. Git tells me Sean wrote that, I'll try to poke him today!

Reply:
Haha you're fine! I was honestly just curious because I didn't see a test case for where we would have to concatenate the line before `;` so I didn't know what it indicated!

Reply:
@nikkiattea TIL https://nagios-plugins.org/doc/guidelines.html#AEN200